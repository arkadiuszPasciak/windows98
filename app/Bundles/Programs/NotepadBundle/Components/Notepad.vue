<template>
  <UIModal
    class="Notepad"
    :title="t('NotepadBundle.title')"
    :resize-window="true"
    width="600px"
    height="450px"
    :modal-state="programStore.modalNotepad"
    @close-modal="closeModal"
  >
    <template #options>
      <UIModalNavigationItem
        data-test="notepad-save-file"
        @click="openSaveModal"
      >
        {{ t('NotepadBundle.save') }}
      </UIModalNavigationItem>

      <FileOpen v-model:text-file-value="textFileValue">
        <UIModalNavigationItem>
          {{ t('NotepadBundle.open') }}
        </UIModalNavigationItem>
      </FileOpen>
    </template>

    <div class="content">
      <FileSave
        class="file-save"
        :modal-state="notepadStore.modalSave"
        :content-value="textareaValue"
        @close-modal="closeSaveModal"
      />
      <UITextarea
        id="notepad-textarea"
        v-model:model-value="textareaValue"
        type="notepad"
        class="textarea"
        label-name=""
        :resize-window="true"
        :disabled="notepadStore.modalSave === true"
      />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, watch } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIModal from '@APP|Bundles/UIModalBundle/Components/UIModal.vue'
  import UIModalNavigationItem from '@APP|Bundles/UIModalBundle/Components/UIModalNavigationItem.vue'
  import UITextarea from '@APP|Bundles/UITextareaBundle/Components/UITextarea.vue'
  import FileOpen from '@APP|Bundles/FileBundle/Components/FileOpen.vue'
  import FileSave from '@APP|Bundles/FileBundle/Components/FileSave.vue'
  import { useProgramStore } from '@APP|Bundles/ProgramBundle/Stores/Program.stores'
  import { useNotepadStore } from '@APP|Bundles/NotepadBundle/Stores/Notepad.stores'

  const { t } = useI18n()

  const programStore = useProgramStore()
  const notepadStore = useNotepadStore()

  const textareaValue = ref<string>(t('NotepadBundle.example'))
  const textFileValue = ref<null | string>(null)

  const closeModal = (): void => {
    programStore.updateNotepadModal(false)
  }

  const openSaveModal = (): void => {
    notepadStore.updateSaveModal(true)
  }

  const closeSaveModal = (): void => {
    notepadStore.updateSaveModal(false)
  }

  watch(textFileValue, async (newValue) => {
    if (newValue) {
      textareaValue.value = newValue
    }
  })
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/NotepadBundle/Styles/Notepad.styles.scss"
/>
