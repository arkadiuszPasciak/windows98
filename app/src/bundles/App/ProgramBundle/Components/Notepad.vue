<template>
  <mf-notepad v-if="programStore.modalNotepad" :on-close-program="closeProgram"></mf-notepad>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP/src/bundles/App/ProgramBundle/Stores/Program.stores"

const programStore = useProgramStore()

const closeProgram = (): void => {
	programStore.updateCalculatorModal(false)
}
</script>
