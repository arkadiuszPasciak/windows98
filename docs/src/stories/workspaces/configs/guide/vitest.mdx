import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

