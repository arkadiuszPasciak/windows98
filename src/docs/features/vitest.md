# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
