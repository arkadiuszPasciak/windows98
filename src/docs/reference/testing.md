# Testing

This documentation provides an overview of the testing process employed in our application. We utilize two powerful testing tools to ensure the reliability and stability of the code: Cypress and Vitest. These tools are employed in different aspects of testing to maximize coverage and maintain code quality.

## [Vitest](https://vitest.dev/)

`@/vite.config.ts` (in the block test)

Vitest is a lightweight and fast testing framework specifically designed for Vite applications. It provides seamless integration with Vite, supporting both JavaScript and TypeScript code. Vitest is particularly well-suited for unit testing, which focuses on verifying the functionality of individual code units, such as functions, classes, or modules.

We use Vitest for `unit testing` our TypeScript code, ensuring that each function or class in our application performs as intended. This approach allows us to isolate potential issues and fix them before they cause problems in the broader application context.

To effectively use Vitest, we create a series of test suites that group related tests, with each test case focusing on a specific unit of code. These test cases cover various scenarios, such as input validation, error handling, and edge cases, to ensure comprehensive coverage of our codebase.
