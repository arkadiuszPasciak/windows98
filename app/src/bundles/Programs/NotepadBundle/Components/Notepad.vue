<template>
  <UIModal
    class="Notepad"
    :title="t('NotepadBundle.title')"
    :resize-window="true"
    width="600px"
    height="450px"
    :modal-state="programStore.modalNotepad"
    @close-modal="closeModal"
  >
    <template #options>
      <UIModalNavigationItem
        data-test="notepad-save-file"
        @click="openSaveModal"
      >
        {{ t('NotepadBundle.save') }}
      </UIModalNavigationItem>

      <FileOpen v-model:text-file-value="textFileValue">
        <UIModalNavigationItem>
          {{ t('NotepadBundle.open') }}
        </UIModalNavigationItem>
      </FileOpen>
    </template>

    <div class="content">
      <FileSave
        class="file-save"
        :modal-state="notepadStore.modalSave"
        :content-value="textareaValue"
        @close-modal="closeSaveModal"
      />
      <UITextarea
        id="notepad-textarea"
        v-model:model-value="textareaValue"
        type="notepad"
        class="textarea"
        label-name=""
        :resize-window="true"
        :disabled="notepadStore.modalSave === true"
      />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP/src/bundles/App/ProgramBundle/Stores/Program.stores"
import { useNotepadStore } from "@APP/src/bundles/Programs/NotepadBundle/Stores/Notepad.stores"
import FileOpen from "@APP|Bundles/FileBundle/Components/FileOpen.vue"
import FileSave from "@APP|Bundles/FileBundle/Components/FileSave.vue"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import UIModalNavigationItem from "@APP|Bundles/UIModalBundle/Components/UIModalNavigationItem.vue"
import UITextarea from "@APP|Bundles/UITextareaBundle/Components/UITextarea.vue"
import { ref, watch } from "vue"
import { useI18n } from "vue-i18n"

const { t } = useI18n()

const programStore = useProgramStore()
const notepadStore = useNotepadStore()

const textareaValue = ref<string>(t("NotepadBundle.example"))
const textFileValue = ref<null | string>(null)

const closeModal = (): void => {
	programStore.updateNotepadModal(false)
}

const openSaveModal = (): void => {
	notepadStore.updateSaveModal(true)
}

const closeSaveModal = (): void => {
	notepadStore.updateSaveModal(false)
}

watch(textFileValue, async (newValue) => {
	if (newValue) {
		textareaValue.value = newValue
	}
})
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/NotepadBundle/Styles/Notepad.styles.scss"
/>
