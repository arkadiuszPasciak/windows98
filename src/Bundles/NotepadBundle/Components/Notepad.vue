<template>
  <UIModal
    class="Notepad"
    :title="t('NotepadBundle.title')"
    :resize-window="true"
    :width="600"
    :height="450"
    :modal-state="programStore.modalNotepad"
    @close-modal="closeModal"
  >
    <template #options>
      <button class="option" type="button" @click="openSaveModal">
        {{ t('NotepadBundle.save') }}
      </button>
    </template>

    <div class="content">
      <FileSave
        class="file-save"
        :modal-state="notepadStore.modalSave"
        @close-modal="closeSaveModal"
      />
      <UITextarea
        id="notepad-textarea"
        type="notepad"
        class="textarea"
        :model-value="modelValue"
        label-name=""
        :resize-window="true"
        :disabled="notepadStore.modalSave === true"
      />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import UITextarea from '@Bundles/UIBundle/Components/UITextarea.vue'
  import FileSave from '@Bundles/FileBundle/Components/FileSave.vue'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import { useNotepadStore } from '@Bundles/NotepadBundle/Stores/Notepad.stores'

  const { t } = useI18n()

  const programStore = useProgramStore()
  const notepadStore = useNotepadStore()

  const modelValue = ref(t('NotepadBundle.example')) as Ref<string>

  const closeModal = (): void => {
    programStore.updateNotepadModal(false)
  }

  const openSaveModal = (): void => {
    notepadStore.updateSaveModal(true)
  }

  const closeSaveModal = (): void => {
    notepadStore.updateSaveModal(false)
  }
</script>

<i18n src="@Bundles/NotepadBundle/Locales/Notepad.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/NotepadBundle/Styles/Notepad.styles.scss';
</style>
