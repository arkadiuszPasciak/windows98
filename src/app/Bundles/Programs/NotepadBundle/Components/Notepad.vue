<template>
  <UIModal
    class="Notepad"
    :title="t('NotepadBundle.title')"
    :resize-window="true"
    width="600px"
    height="450px"
    :modal-state="programStore.modalNotepad"
    @close-modal="closeModal"
  >
    <template #options>
      <UIModalNavigationItem
        data-test="notepad-save-file"
        @click="openSaveModal"
      >
        {{ t('NotepadBundle.save') }}
      </UIModalNavigationItem>

      <FileOpen v-model:text-file-value="textFileValue">
        <UIModalNavigationItem>
          {{ t('NotepadBundle.open') }}
        </UIModalNavigationItem>
      </FileOpen>
    </template>

    <div class="content">
      <FileSave
        class="file-save"
        :modal-state="notepadStore.modalSave"
        :content-value="textareaValue"
        @close-modal="closeSaveModal"
      />
      <UITextarea
        id="notepad-textarea"
        v-model:model-value="textareaValue"
        type="notepad"
        class="textarea"
        label-name=""
        :resize-window="true"
        :disabled="notepadStore.modalSave === true"
      />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref, watch } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIModal from '@Bundles/UIModalBundle/Components/UIModal.vue'
  import UIModalNavigationItem from '@Bundles/UIModalBundle/Components/UIModalNavigationItem.vue'
  import UITextarea from '@Bundles/UITextareaBundle/Components/UITextarea.vue'
  import FileOpen from '@Bundles/FileBundle/Components/FileOpen.vue'
  import FileSave from '@Bundles/FileBundle/Components/FileSave.vue'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import { useNotepadStore } from '@Bundles/NotepadBundle/Stores/Notepad.stores'

  const { t } = useI18n()

  const programStore = useProgramStore()
  const notepadStore = useNotepadStore()

  const textareaValue = ref(t('NotepadBundle.example')) as Ref<string>
  const textFileValue = ref(null) as Ref<null | string>

  const closeModal = (): void => {
    programStore.updateNotepadModal(false)
  }

  const openSaveModal = (): void => {
    notepadStore.updateSaveModal(true)
  }

  const closeSaveModal = (): void => {
    notepadStore.updateSaveModal(false)
  }

  watch(textFileValue, async (newValue) => {
    if (newValue) {
      textareaValue.value = newValue
    }
  })
</script>

<i18n src="@Bundles/NotepadBundle/Locales/Notepad.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/NotepadBundle/Styles/Notepad.styles.scss"
/>
