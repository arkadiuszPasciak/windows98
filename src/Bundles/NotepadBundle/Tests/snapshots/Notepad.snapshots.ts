import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import { createTestingPinia } from '@pinia/testing'
import Notepad from '@Bundles/NotepadBundle/Components/Notepad.vue'

const defaultWrapper = mount(Notepad, {
  global: {
    plugins: [
      createTestingPinia({
        initialState: {
          program: { modalNotepad: true },
        },
      }),
    ],
  },
})

describe('[NotepadBundle]<Snapshots>(Notepad)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
