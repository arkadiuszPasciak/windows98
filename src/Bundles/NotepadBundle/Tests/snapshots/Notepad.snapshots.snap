// Vitest Snapshot v1

exports[`[NotepadBundle]<Snapshots>(Notepad) > renders correctly component 1`] = `
<div
  class="UIModal resize-window move-window Notepad"
  data-v-7632fb2e=""
  data-v-99a75817=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 600px; height: 450px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-test="modal-header-title"
        data-v-99a75817=""
      >
        Notepad
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    <div
      class="UIModalNavigation"
      data-v-99a75817=""
      data-v-9a70d424=""
    >
      
      
      <button
        class="UIModalNavigationItem"
        data-v-0ff8e3f7=""
        data-v-7632fb2e=""
        type="button"
      >
        
        Save
        
      </button>
      <div
        class="FileOpen"
        data-v-7632fb2e=""
        data-v-d06489ec=""
      >
        <div
          class="open-file"
          data-test="file-open-button"
          data-v-d06489ec=""
        >
          
          <button
            class="UIModalNavigationItem"
            data-v-0ff8e3f7=""
            data-v-7632fb2e=""
            type="button"
          >
            
            Open
            
          </button>
          
        </div>
        <input
          class="read-file"
          data-v-d06489ec=""
          type="file"
        />
      </div>
      
      
    </div>
    
    <div
      class="content"
      data-v-7632fb2e=""
    >
      <!--v-if-->
      <div
        class="UITextarea label-position-top type-notepad textarea"
        data-v-7632fb2e=""
        data-v-cbf41378=""
        resize-window="true"
      >
        <!--v-if-->
        <textarea
          class="textarea"
          data-v-cbf41378=""
          id="notepad-textarea"
        />
      </div>
    </div>
    
  </div>
</div>
`;
